include/envoy/upstream/upstream.h

```diff
@@ -45,6 +45,15 @@ class Host : virtual public HostDescription {
     FAILED_EDS_HEALTH = 0x04,
   };
 
+  enum class HealthFailureType {
```

Comment:
Best to rename to `ActiveHealthFailureType` to indicate that this is only for active HC.